domain/service/service.go

```
@@ -803,6 +803,23 @@ func (se *ServiceEndpoint) SetAssignment(aa addressassignment.AddressAssignment)
 	return nil
 }
 
+//SetAddressConfig sets the AddressConfig for the endpoint
+func (s Service) SetAddressConfig(endpointName string, sa servicedefinition.AddressResourceConfig) error {
+	if s.Endpoints == nil {
+		return fmt.Errorf("Service %s has no Endpoints", s.Name)
+	}
+
+	for i := range s.Endpoints {
```

Should return an error if the endpoint does not exist